rest/rest-server/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/RMRestInterface.java

```diff
@@ -163,6 +163,18 @@ NSState createNodeSource(@HeaderParam("sessionid") String sessionId,
             @FormParam("policyType") String policyType, @FormParam("policyParameters") String[] policyParameters,
             @FormParam("policyFileParameters") String[] policyFileParameters) throws NotConnectedException;
 
+    @POST
+    @Path("nodesource/create/recovery")
+    @Produces("application/json")
+    NSState createNodeSource(@HeaderParam("sessionid") String sessionId,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Method has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)